ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/remote/RemoteConnectorExecutor.java

```diff
@@ -189,7 +190,9 @@ default void preparePayloadAndInvoke(
         // override again to always prioritize the input parameter
         parameters.putAll(inputParameters);
         String payload = connector.createPayload(action, parameters);
-        connector.validatePayload(payload);
+        if (!Boolean.parseBoolean(parameters.getOrDefault(SKIP_VALIDATE_MISSING_PARAMETERS, "false"))) {
```

Comment:
I don't think this is a big concern. Just like this blueprint https://github.com/opensearch-project/ml-commons/blob/main/docs/remote_inference_blueprints/cohere_connector_embedding_blueprint.md , user should tune `input_type` and `truncate` based on their use case.

We are not setting the expectation that user can create one connector with a default set of parameters, then it can be used for all use cases.

Reply:
but if connector can be used with dynamic parameters, that would be the best. Let's say for the same connector may be in one agent, `skip_validating_missing_parameters` can be true, but for other agent this can be false?

Reply:
+1, any parameter in the connector can be overwritten later in the "Predict" request. We can treat the parameters in the Connector as default values that are subject to change.

Reply:
Here is our conclusion today: https://github.com/opensearch-project/ml-commons/issues/2748#issuecomment-2259515610